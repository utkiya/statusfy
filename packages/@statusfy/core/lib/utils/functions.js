const { fse, grayMatter, chalk, path } = require("@statusfy/common");

const getIncidentsFromProject = async contentDir => {
  const files = await fse.readdir(contentDir);
  const incidentsList = [];

  for (let i = 0; i < files.length; i++) {
    const f = path.resolve(contentDir, files[i]);
    const ext = path.extname(f);
    const fileName = path.basename(f);

    if (ext === ".md") {
      const fileContent = await fse.readFile(f);
      const { data } = grayMatter.parse(fileContent);

      incidentsList.push({
        value: {
          name: fileName,
          path: f
        },
        name: `${fileName} > ${chalk.yellow(data.title)} (${chalk.green(
          new Date(data.date).toUTCString()
        )})`
      });
    }
  }

  return incidentsList;
};

const generateIncident = (data, content, format) => {
  let matterContent = grayMatter.stringify(content, data, format);

  if (["json", "toml"].includes(format)) {
    matterContent = matterContent.replace(/^---/, `---${format}`);
  }

  return matterContent;
};

module.exports = {
  getIncidentsFromProject,
  generateIncident
};
