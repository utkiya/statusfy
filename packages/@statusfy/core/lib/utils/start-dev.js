const portfinder = require("portfinder");
const bs = require("browser-sync").create();

const { logger, path } = require("@statusfy/common");
const createServer = require("../../server");
const generateConfig = require("../config/generate");

/* eslint-disable require-await */
const start = async () => {
  process.env.NODE_ENV = "development";
  const sourceDir = process.env.STATUSFY_SOURCE_DIR;
  const cliOptions = JSON.parse(process.env.STATUSFY_CLI_OPTIONS);

  // Generate configuration
  const { nuxtConfig, siteConfig } = generateConfig(sourceDir, cliOptions);
  nuxtConfig.mode = cliOptions.ssr ? "universal" : "spa";
  delete nuxtConfig.build.analyze;

  portfinder
    .getPortPromise()
    .then(async port => {
      const serverPort = nuxtConfig.server.port;

      // Update server port
      nuxtConfig.server.port = port;

      // Initialize the server
      await createServer(
        siteConfig,
        nuxtConfig,
        nuxtConfig.server.host,
        nuxtConfig.server.port
      );

      // Create borwser-sync server
      bs.init({
        port: serverPort,
        proxy: `${nuxtConfig.server.host}:${nuxtConfig.server.port}`,
        open: false
      });

      bs.watch(path.join(sourceDir, siteConfig.content.dir, "**/*.md")).on(
        "change",
        bs.reload
      );
    })
    .catch(err => {
      logger.error(err);
    });
};

start();
