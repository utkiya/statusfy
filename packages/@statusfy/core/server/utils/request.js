module.exports = req => {
  const language = req.params.lang || req.app.get("language");
  const page = req.query.page || req.params.lang || 1;

  return {
    language,
    page,
    siteConfig: req.app.get("siteConfig")
  };
};
