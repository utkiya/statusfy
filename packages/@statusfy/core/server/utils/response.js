module.exports = (res, lang) => ({
  json(data) {
    res.setHeader("Content-Language", lang);
    res.json(data);
  },
  notFound(message = "Not Found") {
    res.setHeader("Cache-Control", "no-cache");
    res.status(404).json({
      message
    });
  }
});
