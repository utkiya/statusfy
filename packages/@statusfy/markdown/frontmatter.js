const { grayMatter } = require('@statusfy/common')

module.exports = md => {
  md.frontmatter = md.frontmatter || {}
  md.block.ruler.before('code', 'frontmatter', frontmatter.bind(null, md), {
    alt: []
  })
}

function frontmatter (md, state, start, end, silent) {
  const lines = state.src.split('\n')

  if (start !== 0 || state.blkIndent !== 0 || state.tShift[start] < 0) {
    return false
  }

  // Fist line must start with ---(optional engine name)
  if (!lines[0].match(/^---(toml|json|yaml|yml)?$/)) {
    return false
  }

  // Detect end line number of the front-matter
  let line = 0

  for (let i = 1; i < lines.length; i++) {
    if (lines[i].match(/^---$/)) {
      line = i
      break
    }
  }

  if (line === 0) {
    return false
  }

  // Parse front-matter
  const matterData = grayMatter.parse(state.src)

  md.frontmatter = matterData.data || {}
  state.line = line + 1

  return true
}
