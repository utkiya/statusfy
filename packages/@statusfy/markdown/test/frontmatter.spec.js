import { md, getFragment } from './utils'
import frontmatter from '@/markdown/frontmatter.js'

const mdF = md().use(frontmatter)

describe('frontmatter', () => {
  const containerLabels = ['json', 'toml', 'yaml']

  containerLabels.forEach(label => {
    test(label, async () => {
      const input = await getFragment(`frontmatter-${label}`)
      mdF.render(input)

      expect(mdF.frontmatter).toMatchSnapshot()
    })
  })
})
