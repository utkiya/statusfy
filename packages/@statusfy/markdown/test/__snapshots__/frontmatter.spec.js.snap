// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontmatter json 1`] = `
Object {
  "affectedsystems": Array [
    "conversions",
  ],
  "date": "2018-03-20T02:36:09.000Z",
  "description": "Our Files Conversion System were not responding properly.",
  "modified": "2018-03-20T04:20:09.000Z",
  "resolved": true,
  "severity": "major-outage",
  "title": "Files Conversion Issues",
}
`;

exports[`frontmatter toml 1`] = `
Object {
  "affectedsystems": Array [
    "conversions",
  ],
  "date": 2018-03-20T02:36:09.000Z,
  "description": "Our Files Conversion System were not responding properly.",
  "modified": 2018-03-20T04:20:09.000Z,
  "resolved": true,
  "severity": "major-outage",
  "title": "Files Conversion Issues",
}
`;

exports[`frontmatter yaml 1`] = `
Object {
  "affectedsystems": Array [
    "conversions",
  ],
  "date": 2018-03-20T02:36:09.000Z,
  "description": "Our Files Conversion System were not responding properly.",
  "modified": 2018-03-20T04:20:09.000Z,
  "resolved": true,
  "severity": "major-outage",
  "title": "Files Conversion Issues",
}
`;
